src/fsharp/AccessibilityLogic.fs

```diff
@@ -274,7 +287,10 @@ let GetILAccessOfILPropInfo (ILPropInfo(tinfo, pdef)) =
             | ILMemberAccess.FamilyAndAssembly, _
             | _, ILMemberAccess.FamilyAndAssembly -> ILMemberAccess.FamilyAndAssembly
 
-            | _ -> ILMemberAccess.Private
+            | ILMemberAccess.Private, _
+            | _, ILMemberAccess.Private -> ILMemberAccess.Private
+
+            | _ -> ILMemberAccess.CompilerControlled
```

Comment:
```suggestion
            | ILMemberAccess.CompilerControlled, _
            | _, ILMemberAccess.CompilerControlled -> ILMemberAccess.CompilerControlled
```

this way, an addition will trigger a warning on this match expression lacking new cases.